apps/backend/src/websocket/connection_manager.py

```
@@ -0,0 +1,249 @@
+        
+        connection = self.connections[client_id]
+        
+        # Require authentication for subscriptions
+        if not connection.authenticated:
+            await connection.send_error("AUTH_REQUIRED", "Authentication required for subscriptions")
+            return
+        
+        connection.update_subscriptions(subscription_msg.action, subscription_msg.topics)
+        # Subscription updated
+    
+    async def handle_heartbeat(self, client_id: str, heartbeat_msg: HeartbeatMessage):
```

Ruff (0.12.2)

Unused method argument: `heartbeat_msg`

(ARG002)